src/backend/common/SparseArray.cpp

```diff
@@ -127,15 +126,18 @@ SparseArray<T> createHostDataSparseArray(const af::dim4 &_dims, const dim_t nNZ,
                                          const int *const _rowIdx,
                                          const int *const _colIdx,
                                          const af::storage _storage) {
-    return SparseArray<T>(_dims, nNZ, _values, _rowIdx, _colIdx, _storage,
-                          false);
+    return SparseArray<T>(_dims, nNZ, const_cast<T *>(_values),
+                          const_cast<int *>(_rowIdx),
+                          const_cast<int *>(_colIdx), _storage, false);
```

Comment:
What's the difference between earlier and now in this constructor ?

Reply:
The older SparseArray was const. this was a problem because we accept device functions and those can be modified by the internals. Moving them to non-const SparseArray requires this cast.